### DISCUSSION TITLE
[Error] make sure you have FULL CONTROL at the source site: {URL}

### DISCUSSION THREAD

**Discussion by odross87** [Zerg00s/FlowPowerAppsMigrator#50] [Q&A]
Hello,

I am currently trying to use the migrator and I'm encountering the following error after inputting my credentials: "[Error] make sure you have FULL CONTROL at the source site:". I have already confirmed that the account I'm using has Full Control access.

Has anyone else experienced this issue before? If so, do you have any suggestions on how to resolve it?

Thank you in advance for any assistance you can provide.

_

[Error] make sure you have FULL CONTROL at the source site: https://xxxxxxxxxxxxxxxxx
If you already have enough permissions, try running the script with CLEAR_CREDENTIALS_CACHE set to True
Invoke-PnPSPRestMethod : Error en el servidor remoto: (403) Prohibido.
En C:\Users\xxxxxxxxxxxxx\xxxxxxxx\FlowPowerAppsMigrator\GenerateInitialMapping.ps1: 20 Carácter: 17
+ ... $Response = Invoke-PnPSPRestMethod -Url "$SOURCE_SITE_URL/_api/web/cu ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Invoke-PnPSPRestMethod], WebException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Admin.InvokeSPRestMethod
Presione una tecla para continuar . . .

_

**Comment by Zerg00s** [comment]
Hi @odross87,

Can we do a test, please? I want to try to isolate the issue.

## Install the legacy PnP PowerShell module:
Open Powershell and run this command
```
Install-Module SharePointPnPPowerShellOnline -Scope CurrentUser
```

## Test Connection to SharePoint Online:

```
Connect-PnPOnline -Url <Your SharePoint site> -UseWebLogin -WarningAction Ignore
Get-PnPList -Includes Views, Fields, DefaultView
```

If you do this, what is the result? Does it also throw an error?

**Reply by odross87** [reply]
Let me try tomorrow and i'll text you. 

Thank you in advance

**Comment by Zerg00s** [comment]
I've just created a quick Wiki page on a few things to try. Can you have a look at this, please? https://github.com/Zerg00s/FlowPowerAppsMigrator/wiki/How-to-deal-with-the-%22Make-sure-you-have-FULL-CONTROL-at-the-source-site%22-Issue

If you are the lucky one who has SharePoint Tenant Admin role, then you can create a SharePoint-only app and use it for authentication.
